### PR TITLE
Add version to Galley MCP dial out metadata.

### PR DESCRIPTION
**Pull request by jeffmendoza**
#13069

@ja30278 @panywei

### CODE REVIEW

File: galley/pkg/server/callout.go [hunk 1]
```diff
@@ -25,8 +25,11 @@ import (
 	mcp "istio.io/api/mcp/v1alpha1"
 	"istio.io/istio/galley/pkg/authplugins"
 	"istio.io/istio/pkg/mcp/source"
+	"istio.io/istio/pkg/version"
 )
 
+const versionMetadataKey = "galley-version"
```
Comment (ozevren) [inline galley/pkg/server/callout.go]: I don't think "galley-version" is the right name for this. Call it config.source.version?

Comment (jeffmendoza) [reply galley/pkg/server/callout.go]: It's the version that gets baked into galley when the binary is built. I don't want to call it "istio version" as it doesn't necessarily represent the version of the whole mesh. If we consider "config source" an analog for galley, then that works for me.

**Comment by jeffmendoza** [comment]
@ozevren It seems like an integration test for this might be too disruptive right now. server_sink.go does not capture the metadata from the stream context. It would need to do this, the propagate this somehow, Likely through the sink.Change struct that is passed to the Updater. The we can update the InMemoryUpdater to allow grabbing the version.

**Pull request merged**